File: spec/javascript/test_nested_types2.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('NestedTypes2', 'src/fixed_struct.bin', function(r) {
  assert.equal(r.one.typedAtRoot.valueB, 80);

  assert.equal(r.one.typedHere1.valueC, 65);

  assert.equal(r.one.typedHere1.typedHere.valueD, 67);
  assert.equal(r.one.typedHere1.typedParent.valueCc, 75);
  assert.equal(r.one.typedHere1.typedRoot.valueB, 45);

  assert.equal(r.one.typedHere2.valueCc, 49);

  assert.equal(r.two.valueB, -1);
});

File: spec/python/test_nested_types2.py
import unittest

from nested_types2 import NestedTypes2

class TestNestedTypes2(unittest.TestCase):
    def test_nested_types2(self):
        r = NestedTypes2.from_file("src/fixed_struct.bin")

        self.assertEquals(r.one.typed_at_root.value_b, 80)

        self.assertEquals(r.one.typed_here1.value_c, 65)

        self.assertEquals(r.one.typed_here1.typed_here.value_d, 67)
        self.assertEquals(r.one.typed_here1.typed_parent.value_cc, 75)
        self.assertEquals(r.one.typed_here1.typed_root.value_b, 45)

        self.assertEquals(r.one.typed_here2.value_cc, 49)

        self.assertEquals(r.two.value_b, -1)
